src/main/java/io/cdap/plugin/servicenow/sink/ServiceNowSinkConfig.java

```diff
@@ -0,0 +1,304 @@
+
+  @Name(ServiceNowConstants.PROPERTY_TABLE_NAME)
+  @Macro
+  @Description("The name of the ServiceNow table to which data needs to be inserted.")
+  private String tableName;
+
+  @Name(ServiceNowConstants.PROPERTY_OPERATION)
+  @Macro
+  @Description("The type of the operation to be performed in ServiceNow tables. Note, for Update operation, sys_id " +
+    "must be present.")
+  private String operation;
+
+  @Name(ServiceNowConstants.PROPERTY_MAX_RECORDS_PER_BATCH)
+  @Macro
+  @Description("No. of requests that will be sent to ServiceNow Batch API as a payload. Default value is 200. Rest " +
+    "API property in Transaction quota section \"REST Batch API request timeout\" should be increased to use higher " +
+    "records in a batch. By default this property has a value of 30 sec which can handle approximately 200 records " +
+    "in a batch. To use a bigger batch size, set it to a higher value. ")
+  private Long maxRecordsPerBatch;
+
+  @Name(ServiceNowConstants.NAME_SCHEMA)
+  @Macro
+  @Nullable
+  @Description("The schema of the table to read.")
+  private String schema; 
+
+  /**
+   * Constructor for ServiceNowSourceConfig object.
+   *
+   * @param referenceName The reference name
+   * @param clientId The Client Id for ServiceNow
+   * @param clientSecret The Client Secret for ServiceNow
+   * @param restApiEndpoint The rest API endpoint for ServiceNow
+   * @param user The user id for ServiceNow
+   * @param password The password for ServiceNow
+   * @param tableName The table name
+   * @param operation The type of operation to be performed
+   * @param maxRecordsPerBatch The maximum number of records per batch
+   */
+  public ServiceNowSinkConfig(String referenceName, String clientId, String clientSecret, String restApiEndpoint,
+                              String user, String password, String tableName,
+                              String operation, Long maxRecordsPerBatch) {
+    super(referenceName, clientId, clientSecret, restApiEndpoint, user, password);
+    this.tableName = tableName;
+    this.operation = operation;
+    this.maxRecordsPerBatch = maxRecordsPerBatch;
+  }
+
+  @Nullable
+  public String getTableName() {
+    return tableName;
+  }
+
+  public String getOperation() {
+    return operation;
+  }
+
+  public Long getMaxRecordsPerBatch() {
+    return this.maxRecordsPerBatch == null ? 0 : this.maxRecordsPerBatch;
+  }
+    
+  /**
+   * Validates {@link ServiceNowSinkConfig} instance.
+   */
+  public void validate(FailureCollector collector) {
+    super.validate(collector);
+    validateMaxRecordsPerBatch(collector);
+    validateTable(collector);
+  }
+
+  private void validateMaxRecordsPerBatch(FailureCollector collector) {
+    if (containsMacro(ServiceNowConstants.PROPERTY_MAX_RECORDS_PER_BATCH)) {
+      return;
+    }
+
+    if (Objects.isNull(maxRecordsPerBatch)) {
+      collector.addFailure("Max records per batch must be specified.", null)
```

Comment:
In line 110 you're handling maxRecoresPerBatch == null. Can it be null?

Reply:
Yes, it can be null in case a macro is provided by the user.
Removed the additional `null` check at line 110. `maxRecoresPerBatch == null.`, as it is not required. 5d38574b7fa5eb4560278961cc05d061f5c47efb